python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -397,18 +436,56 @@ def run(self, inputWarps, skyInfo, **kwargs):
                 warp.mask.array = artifactMask.array
                 del artifactMask
 
-            # Pre-process the warp before coadding.
-            # Each Warp that goes into a coadd will typically have an
-            # independent photometric zero-point. Therefore, we must scale each
-            # Warp to set it to a common photometric zeropoint.
             if self.config.do_scale_zero_point:
-                self.scale_zero_point.run(exposure=warp, dataRef=warpRef)
+                # Each Warp that goes into a coadd will typically have an
+                # independent photometric zero-point. Therefore, we must scale
+                # each Warp to set it to a common photometric zeropoint.
+                imageScaler = self.scale_zero_point.run(exposure=warp, dataRef=warpRef).imageScaler
+                zero_point_scale_factor = imageScaler.scale
+                self.log.debug(
+                    "Scaled the warp %s by %f to match zero points", warpRef.dataId, zero_point_scale_factor
+                )
+            else:
+                zero_point_scale_factor = 1.0
+                if "BUNIT" not in warp.metadata:
+                    raise ValueError(f"Warp {warpRef.dataId} has no BUNIT metadata")
+                if warp.metadata["BUNIT"] != "nJy":
+                    raise ValueError(
+                        f"Warp {warpRef.dataId} has BUNIT {warp.metadata['BUNIT']}, expected nJy"
+                    )
 
             # Coadd the warp onto the cells it completely overlaps.
             edge = afwImage.Mask.getPlaneBitMask(["NO_DATA", "SENSOR_EDGE"])
             reject = afwImage.Mask.getPlaneBitMask(["CLIPPED", "REJECTED"])
             removeMaskPlanes(warp.mask, self.config.remove_mask_planes, self.log)
 
+            # Compute the weight for each CCD in the warp from the visitSummary
+            # or from the warp itself, it not provided. Computing the weight
+            # from the warp is not recommended, and in that case we compute one
+            # weight per warp and not bother with per-detector weights.
+            weights: Mapping[int, float] = {}  # Mapping from detector to weight.
+            full_ccd_table = warp.getInfo().getCoaddInputs().ccds
+
+            if visitSummaryRef:
+                assert visitSummaryRef.dataId["visit"] == warpRef.dataId["visit"]
+                visitSummary = visitSummaryRef.get()
+                for detector in full_ccd_table["ccd"]:
+                    visitSummaryRow = visitSummary.find(detector)
+                    mean_variance = visitSummaryRow["meanVar"]
+                    mean_variance *= zero_point_scale_factor**2
+                    if warp.metadata.get("BUNIT", None) == "nJy":
+                        mean_variance *= visitSummaryRow.photoCalib.getCalibrationMean() ** 2
```

Comment:
The need for the `PhotoCalib` object here means we can't use `ccdVisitTable` in place of `finalVisitSummary`; unless we put this info there

Reply:
I think loading the visit summary here is probably better than either calibrating the summary stats quantities or putting the calibration mean into the CCD-visit table.